app/controllers/conversions/academy_urn_controller.rb

```diff
@@ -0,0 +1,58 @@
+class Conversions::AcademyUrnController < ApplicationController
+  rescue_from Api::AcademiesApi::Client::NotFoundError, with: :not_found_error
+
+  def edit
+    @project = Project.find(params[:project_id])
+    authorize @project
+  end
+
+  def check
+    @project = Project.find(params[:project_id])
+    authorize @project
+
+    if academy_urn.blank?
+      @project.errors.add(:academy_urn, :blank)
+      return render :edit
+    end
+
+    @project.assign_attributes(academy_urn: academy_urn)
+    unless @project.valid?
+      @project.errors.add(:academy_urn, :invalid_urn)
+      return render :edit
+    end
+
+    result = fetch_academy_details(academy_urn)
+    @academy_urn = academy_urn
+    @establishment = result
+
+    render "confirm"
+  end
+
+  def update
```

Comment:
I think we need to give this a specific name so we can allow this update to happen when the user is not `assigned_to`, right now I can get stuck trying to update the academy urn for a project I am not assigned to - if we call this `update_academy_urn` or something, we can add a specific permission for it?